libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -293,31 +214,42 @@ void AP_Proximity_RPLidarA2::get_readings()
             } else {
                 // unknown descriptor.  Ignore it.
             }
-            consume_bytes(sizeof(_descriptor));
             break;
 
         case State::AWAITING_DEVICE_INFO:
-            if (_byte_count < sizeof(_payload.device_info)) {
+            if (_uart->available() < sizeof(_payload.device_info)) {
+                return;
+            }
+            if ((_read_result = _uart->read(&_payload[0], sizeof(_payload.device_info))) > 0) {
+                _bytes_read += _read_result;
+            } else {
                 return;
```

Comment:
This return means that in the case you don't get the number you're after you'll continue to loop endlessly.  Again, you've asked how many bytes are available, if you don't get the number you expect then something drastic is probably in order.

I should say that's not an impossibility, BTW.  serial-port-passthrough and other hijinks can do strange things.

Reply:
No, we will no endlessly loop here, because there is a timeout in the driver that triggers a reset.